gnosis/safe-react
# Fix ignored safeTxGas from app

Closes #2091 

Safe Apps have a property that can be used to suggest the necessary safeTxGas. We were completely ignoring it.

**The expected behavior should be:**
 - Keep the suggested safeTxGas from the app.
 - If the safeTxGas suggested by the app is lower than our estimation we just simply swap it by our estimation.

Functions and test from `gas` file deleted on this PR are legacy functions of safeTxGas calculation
 
**How to test:**
 - You can add this test app that allows you to set a custom safeTxGas:
 https://ipfs.io/ipfs/QmfMq8NgxdDfai1qKv7bGbT8DegYX4mNzmpi9AkY1VJZBS/

- safeTxGas value can be checked in "Advanced Parameters" section
- If the value is big enough it should use the suggested safeTxGas
- If the value is set to a very low value in the safe app form, let's say 10000, safeTxGas will be our backend estimation
- safeTxGas is correctly sent to Metamask or any other wallet and transactions can be executed without issues
- In general other transactions aren't affected by this change (send funds, settings update...)
